### PR TITLE
Dev ccrma ecasound

### PR DESCRIPTION
**Pull request by cchafe**
this is the first stage of integrating hubPatch -p6 (panning / reverb feature which is already running 24/7 for Stanford's ensembles)

### CODE REVIEW

**Comment by ntonnaett** [comment]
Tell me when it's ready.

I see after this main is not in front of dev anymore. Maybe I'll make another PR for that.

**Comment by cchafe** [comment]
ok, let's hold off until some new experimentation I'm working on is done
should I remove the PR until then?

**Comment by ntonnaett** [comment]
No, it's not a problem. We could make a draft out of it, if you wish.

**Comment by cchafe** [comment]
Cool -- didn't know that feature. It's now a "draft" PR

File: src/JMess.cpp [hunk 1]
```diff
@@ -35,6 +35,13 @@
 
 #include "JackTrip.h"
 #include "jacktrip_globals.h"
+#include <QDebug>
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER "SuperCollider"
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER_IN ":in_"
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER_OUT ":out_"
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER_PANSTEREO "panpot9toStereo"
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER_FREEVERBSTEREO "freeverbStereo"
+#define HARDWIRED_AUDIO_PROCESS_ON_SERVER_ECASOUND "ecasound"
```
Comment (ntonnaett) [inline src/JMess.cpp:44]: Prefer using constexpr char[] and putting it in the namespace where it's used.

File: src/JMess.cpp [hunk 2]
```diff
@@ -352,6 +372,211 @@ void JMess::connectTUB(int /*nChans*/)
...
+// panning and reverb effectively acting like FULLMIX above
+// used by Stanford ensembles, Fall 2020
+// this gets run on the ensemble's hub server with
+// ./jacktrip -S -p6
+// 6=stereo room w/pan9+reverb and connects a variable set of client jacktrips
+// to a known hardwired audio process with known hardwired audio port names
+// when clients connect / disconnect dynamically this just runs through the
+// audio connection sequence bruteforce at every new connection change
+// panning input has 9 slots in the stereo field
+#define NPANINCHANS 9
```
Comment (ntonnaett) [inline src/JMess.cpp:386]: Prefer constexpr

File: src/plc.cpp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#include "plc.h"
+
+PLC::PLC()
```
Comment (ntonnaett) [inline src/plc.cpp:3]: If you don't need a custom constructor don't define one. Your class will become trivially constructable, copyable and movable.

**Comment by cchafe** [comment]
I'm sketching out and eventually testing PLC integration in this draft. Maybe best to remove it while that's in progress? (like the new class template just added -- very early stage)

**Comment by cchafe** [comment]
I'm going to remove -- too early stage and even though a draft it still gets travis'd every commit
Don't want to waste others time or $$ (yes, we just got a bill from travis!)

**Pull request closed without merging**